src/client/components/global_header/Header.tsx

```
@@ -139,17 +139,22 @@ export class Header extends Component<Props, State> {
     }
     const headers = {
       Authorization: token,
+      Accept: 'application/json',
     };
     fetch(Runtime.getConfig().service_routes.auth + '/logout', {
       method: 'POST',
       headers,
     })
-      .then(() => {
-        // Remove the cookie
-        removeCookie('kbase_session');
-        // Redirect to the legacy signed-out page
-        window.location.href =
-          Runtime.getConfig().host_root + '/#auth2/signedout';
+      .then(resp => {
+        if (resp.ok) {
```

Added a check for failed requests. This aligns failure modes that were causing different behavior on CI and prod* when the sign out button was clicked. It prevents purging the local tokens unless the logout request succeeds.

* Prod was failing on CORS due to a 404 on the OPTIONS call (as the endpoint DNE), which was caught. But, CI was failing with a 404 on POST (because it wasn't making a cross-origin request to the service) and this was not caught, giving the appearance that logout was succeeding while it was silently failing.